### DISCUSSION TITLE
Stable Preview Deployment URLs

### DISCUSSION THREAD

**Discussion by mrtom** [vercel/community#4613] [Help] [answered]
### Summary

Hi.

According to [this guide](https://vercel.com/guides/how-to-alias-a-preview-deployment-using-the-cli), 

> Each preview deployment has a unique URL. The latest Preview Deployment is automatically aliased to a URL, ensuring you'll have a stable URL pointing to the last Preview Deployment.

However, I can't see a stable URL for preview deployments anywhere in my project.

In the Domain settings for my app I have two domains setup - foo.vercel.app, setup by Vercel, and myapp.com, setup by me. Both of these work, and deploy the latest code pushed to my main branch (in GitHub) automatically.

The text on the Domains settings page states:

> These domains are assigned to your Production Deployments

So I don't think I should be configuring something here for preview deployments.

If I look at the deployments for my project, and filter for preview deploys, I see one for each commit I pushed to main. But they all have unique URLs, eg. app-name-unique-slug-org-name.vercel.app.

I'm wondering how to find (or configure) a stable URL for the latest deploy to preview for my app?

Many thanks!

### Steps to Reproduce

Nothing to add here

**Comment by amyegan** [comment, marked as answer]
There are a few options. You can use the automatically provided alias URLs for your branch. These are different depending on whether you use the automatic Git integration or the CLI to deploy, but they each use a predictable formula. You can use the CLI or API to alias a preview deployment. Or you can assign a domain in project settings that will be applied to the latest deployment from a specific branch.

- [Generated URL with Git branch](https://vercel.com/docs/deployments/generated-urls#url-with-git-branch)
- [Generated URL with Vercel CLI](https://vercel.com/docs/deployments/generated-urls#url-with-vercel-cli)
- [Assign an Alias via Vercel CLI](https://vercel.com/docs/cli/alias#)
- [Assign an Alias via Vercel REST API](https://vercel.com/docs/rest-api/endpoints#assign-an-alias)
- [Assigning a Domain to a Git Branch](https://vercel.com/docs/projects/domains/assign-domain-to-a-git-branch#)

**Comment by mrtom** [comment]
Thanks for the response @amyegan.

Just so I'm clear:

> You can use the automatically provided alias URLs for your branch. These are different depending on whether you use the automatic Git integration or the CLI to deploy, but they each use a predictable formula.

I think you're saying here that this doesn't solve my use case. I'm looking for a stable URL, and I believe you are saying these will be unstable, but predictable. Is that correct?

> You can use the CLI or API to alias a preview deployment

Great, this sounds useful. Just to confirm (because I've tried and am getting an error). Do I need to set the domain in the Domains section in the Project Settings first, and then run the alias command?

I have tried the following:

```
vercel alias set content-moderation-8ekye5jgz-lime-and-soda.vercel.app console.preview.contentwarden.com
```

`console.preview.contentwarden.com` is setup correctly in my project settings (but with a 'production' tag rather than a preview one), but the CLI is giving the following error:

```
Error: You don't have access to the domain console.preview.contentwarden.com under temelliott
```

> Or you can assign a domain in project settings that will be applied to the latest deployment from a specific branch.

Presumably, however, if I go down this route I would have a production deploy on that branch rather than a preview deploy. Is that correct?

Many thanks!

**Comment by mrtom** [comment]
I suspect the error is caused by having the Vercel CLI logged in to my personal vercel account and not the organization that the project is in?

**Comment by mrtom** [comment]
OK, yes that was it.

`vercel switch team_name` saved the day.

**Comment by mrtom** [comment]
And for those following along at home, `vercel alias set preview_url mydomain.com` has saved the day.

Now to wire that up to CI 😄
